packages/gatsby/src/redux/run-sift.js

```diff
@@ -181,13 +182,27 @@ const getBucketsForFilters = (filters, nodeTypeNames, filtersCache) => {
   // Fail fast while trying to create and get the value-cache for each path
   let every = filters.every((filter /*: DbQuery*/) => {
     let cacheKey = createTypedFilterCacheKey(nodeTypeNames, filter)
-    getBucketsForQueryFilter(
-      cacheKey,
-      filter,
-      nodeTypeNames,
-      filtersCache,
-      filterCaches
-    )
+    if (filter.type === `query`) {
+      // (Let TS warn us if a new query type gets added)
+      const q /*: IDbQueryQuery */ = filter
```

Comment:
Flow supports a "legacy" syntax like that, multi-line comment starting with colon, which it would read just like not having that comment.

Not that this matters much. The main point is to make it easier to move this file to TS when the time comes. I'll want to keep this typed. If another type of query gets added, TS will complain here. So after this file becomes TS this would be `const q: IDbQueryQuery = filter`, with little/less effort for the migration.